src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -1820,8 +1820,8 @@ public void writeStyle( StyleInfo info, Style style ) throws IOException {
      */
     public void writeStyle( StyleInfo info, Style style, boolean format) throws IOException {
         synchronized ( styleCache ) {
-            File styleFile = dataDir().findOrCreateStyleSldFile(info);
-            BufferedOutputStream out = new BufferedOutputStream( new FileOutputStream( styleFile ) );
+            Resource styleFile = dataDir().style(info);
+            BufferedOutputStream out = new BufferedOutputStream( styleFile.out() );
```

Comment:
Unwanted whitespace inside brackets. I am certain I know who did this (Justin the persistent style breaker did a huge amount of catalog work; I do not want to at-tag him), but please fix style as you go.  :-)